sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4109,48 +4122,55 @@ pi_result piEnqueueEventsWait(pi_queue Queue, pi_uint32 NumEventsInWaitList,
   PI_ASSERT(Queue, PI_INVALID_QUEUE);
   PI_ASSERT(Event, PI_INVALID_EVENT);
 
-  _pi_ze_event_list_t TmpWaitList;
-  if (auto Res = TmpWaitList.createAndRetainPiZeEventList(NumEventsInWaitList,
-                                                          EventWaitList, Queue))
-    return Res;
+  if (EventWaitList) {
```

Comment:
The change here is just a restructuring to put command-list allocation under this condition, i.e. when it is really needed.

Reply:
Thanks for the note which helped me a lot to review the change.
Nice catch here for the unnecessary command-list allocation.